### PR TITLE
MES-4386: Screens for Category A Mod1

### PR DESCRIPTION
**Pull request by szabikr**
## Description

All the screens which are needed for category A Mod1.

The changes have been reviewed already. But look out for any missing TODOs or any names that are incorrect.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by edwardwilson — APPROVED** [review]
Checked for TODO's regarding moving to AMod1 types again and all looks good!

File: src/pages/back-to-office/cat-a-mod1/_tests_/back-to-office.cat-a-mod1.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+import { DeviceProvider } from '../../../../providers/device/device';
+import { DeviceProviderMock } from '../../../../providers/device/__mocks__/device.mock';
+import { InsomniaMock } from '../../../../shared/mocks/insomnia.mock';
+import { ScreenOrientationMock } from '../../../../shared/mocks/screen-orientation.mock';
+import { MockComponent } from 'ng-mocks';
+import { PracticeModeBanner } from '../../../../components/common/practice-mode-banner/practice-mode-banner';
+import { By } from '@angular/platform-browser';
+import { of } from 'rxjs/observable/of';
+
+describe('BackToOfficePage', () => {
```
Comment (TomBillingtonUK) [inline src/pages/back-to-office/cat-a-mod1/_tests_/back-to-office.cat-a-mod1.page.spec.ts]: Need Category here to make working out which tests failed easier

File: src/pages/back-to-office/cat-a-mod1/back-to-office.cat-a-mod1.page.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+
+<ion-header>
+    <practice-mode-banner *ngIf="isPracticeMode"></practice-mode-banner>
```
Comment (TomBillingtonUK) [inline src/pages/back-to-office/cat-a-mod1/back-to-office.cat-a-mod1.page.html]: Can be removed as this isn’t a Practicable Base Page

File: src/pages/debrief/cat-a-mod1/__tests__/debrief.cat-a-mod1.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        StoreModule.forRoot({
+          tests: () => ({
+            currentTest: {
+              slotId: '123',
+            },
+            testStatus: {},
+            startedTests: {
+              123: {
+                testSlotAttributes,
+                category: TestCategory.BE,
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-a-mod1/__tests__/debrief.cat-a-mod1.page.spec.ts:98]: Need a todo to update mock data

File: src/pages/non-pass-finalisation/cat-a-mod1/__tests__/non-pass-finalisation.cat-a-mod1.page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+describe('NonPassFinalisationCatAMod1Page', () => {
+  let fixture: ComponentFixture<NonPassFinalisationCatAMod1Page>;
+  let component: NonPassFinalisationCatAMod1Page;
+  let store$: Store<StoreModel>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        NonPassFinalisationCatAMod1Page,
+        MockComponent(PracticeModeBanner),
```
Comment (TomBillingtonUK) [inline src/pages/non-pass-finalisation/cat-a-mod1/__tests__/non-pass-finalisation.cat-a-mod1.page.spec.ts]: Not needed as there is no banner on page

File: src/pages/office/cat-a-mod1/__tests__/office.cat-a-mod1.page.spec.ts [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+        ComponentsModule,
+        StoreModule.forRoot({
+          tests: () => ({
+            currentTest: {
+              slotId: '123',
+            },
+            testStatus: {},
+            startedTests: {
+              123: {
+                category: TestCategory.BE,
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/__tests__/office.cat-a-mod1.page.spec.ts:103]: Need todo to update mock data

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 6]
```diff
@@ -0,0 +1,607 @@
...
+      displayDrivingFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, data]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:323]: Needs updating

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 7]
```diff
@@ -0,0 +1,607 @@
...
+      displaySeriousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, data]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getSeriousFaultsList(data, TestCategory.BE),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:337]: Needs updating

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 8]
```diff
@@ -0,0 +1,607 @@
...
+      displayDangerousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, data]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:351]: Needs a todo

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 9]
```diff
@@ -0,0 +1,607 @@
...
+        select(getETAFaultText),
+      ),
+      ecoFaults$: currentTest$.pipe(
+        select(getTestData),
+        select(getEco),
+        select(getEcoFaultText),
+      ),
+      dangerousFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.faultSummaryProvider.getDangerousFaultsList(data, TestCategory.BE)),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:393]: Needs todo

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 10]
```diff
@@ -0,0 +1,607 @@
...
+      ),
+      seriousFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.faultSummaryProvider.getSeriousFaultsList(data, TestCategory.BE)),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:399]: Needs todo

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 11]
```diff
@@ -0,0 +1,607 @@
...
+      ),
+      drivingFaults$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.faultSummaryProvider.getDrivingFaultsList(data, TestCategory.BE)),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:405]: Needs todo

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts [hunk 12]
```diff
@@ -0,0 +1,607 @@
...
+      ),
+      drivingFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        map(data => this.faultCountProvider.getDrivingFaultSumCount(TestCategory.BE, data)),
```
Comment (TomBillingtonUK) [inline src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:411]: Needs todo

**Review by TomBillingtonUK — APPROVED** [review]
Looks good put comments on the things I spotted

File: src/pages/back-to-office/cat-a-mod1/_tests_/back-to-office.cat-a-mod1.page.spec.ts (hunk 1 above)
Comment (szabikr) [reply src/pages/back-to-office/cat-a-mod1/_tests_/back-to-office.cat-a-mod1.page.spec.ts]: Added

File: src/pages/back-to-office/cat-a-mod1/back-to-office.cat-a-mod1.page.html (hunk 2 above)
Comment (szabikr) [reply src/pages/back-to-office/cat-a-mod1/back-to-office.cat-a-mod1.page.html]: Removed

File: src/pages/debrief/cat-a-mod1/__tests__/debrief.cat-a-mod1.page.spec.ts (hunk 3 above)
Comment (szabikr) [reply src/pages/debrief/cat-a-mod1/__tests__/debrief.cat-a-mod1.page.spec.ts:98]: Added todo

File: src/pages/non-pass-finalisation/cat-a-mod1/__tests__/non-pass-finalisation.cat-a-mod1.page.spec.ts (hunk 4 above)
Comment (szabikr) [reply src/pages/non-pass-finalisation/cat-a-mod1/__tests__/non-pass-finalisation.cat-a-mod1.page.spec.ts]: Removed

File: src/pages/office/cat-a-mod1/__tests__/office.cat-a-mod1.page.spec.ts (hunk 5 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/__tests__/office.cat-a-mod1.page.spec.ts:103]: Added

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 6 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:323]: I'll add todos instead. We have had some issues with just choosing the defaults on these functions

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 7 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:337]: Updated

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 8 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:351]: Added

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 9 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:393]: Added

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 10 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:399]: Added

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 11 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:405]: Added

File: src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts (hunk 12 above)
Comment (szabikr) [reply src/pages/office/cat-a-mod1/office.cat-a-mod1.page.ts:411]: Added

**Pull request merged**